packages/protocol/contracts/common/Accounts.sol

```diff
@@ -857,6 +922,56 @@ contract Accounts is
     return (sizes, data);
   }
 
+  /**
+   * @notice Getter for the storage roots of multiple accounts.
+   * @param accountsToQuery The addresses of the accounts to get the storage roots for.
+   * @return (
+   *          data - all strings concatenated
+   *          rootLengths[] - the length of each string in `data` in bytes
+   *          accountRootNumbers[] - the number of storage roots per account
+   *         )
+   */
+
+  function batchGetOffchainStorageRoots(address[] calldata accountsToQuery)
```

Comment:
maybe lacking context but don't see why this is useful
I suppose to reduce number of round trips but the RPC supports batching calls too

Reply:
I believe avoiding round trips was the main goal with this. If this is supported out-of-contract that would be ideal (especially given how complicated this function ended up being), let's discuss this with the wallet team that would be consuming this.